### PR TITLE
Reader comments: only trust comment content from the API

### PR DESCRIPTION
**Pull request by bluefuton**
We currently render all Reader comment content using `dangerouslySetInnerHtml`, including previews.

To benefit from React's in-built escaping, I've created a new `PostCommentContent` component which only uses `dangerouslySetInnerHtml` on trusted comment content received from the REST API.

### CODE REVIEW

File: client/reader/comments/post-comment-content.jsx [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+const PostCommentContent = React.createClass( {
+	propTypes: {
+		content: PropTypes.string.isRequired,
+		state: PropTypes.string.isRequired,
+	},
+
+	render() {
+		// Don't trust comment content unless it was provided by the API
+		if ( this.props.state !== CommentConstants.state.COMPLETE ) {
+			<div className="comment__content">{ this.props.content }</div>
```
Comment (blowery) [inline client/reader/comments/post-comment-content.jsx]: i think you need a `return` here

File: client/reader/comments/post-comment-content.jsx [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import React, { PropTypes } from 'react';
+import CommentConstants from 'lib/comment-store/constants';
+
+const PostCommentContent = React.createClass( {
+	propTypes: {
```
Comment (blowery) [inline client/reader/comments/post-comment-content.jsx]: This should pull in PureRenderMixin

File: client/reader/comments/post-comment-content.jsx (hunk 1 above)
Comment (bluefuton) [inline client/reader/comments/post-comment-content.jsx]: Oops. Yes, that'd help... :)

File: client/reader/comments/post-comment-content.jsx (hunk 2 above)
Comment (bluefuton) [inline client/reader/comments/post-comment-content.jsx]: :+1:

File: client/lib/comment-store/comment-store.js [hunk 3]
```diff
@@ -132,7 +132,7 @@ function confirmComment( siteId, postId, commentPlaceholderId, confirmedComment
 	}
 
 	// Replace the comment with the new one and set the state to complete
-	confirmedComment.state = States.CONFIRMED;
+	confirmedComment.state = States.COMPLETE;
```
Comment (blowery) [inline client/lib/comment-store/comment-store.js]: Can we add a test for this while we're here?

File: client/reader/comments/post-comment-content.jsx [hunk 4]
```diff
@@ -0,0 +1,26 @@
+import React, { PropTypes } from 'react';
```
Comment (blowery) [inline client/reader/comments/post-comment-content.jsx]: needs to be 'react/addons'

File: client/lib/comment-store/comment-store.js (hunk 3 above)
Comment (bluefuton) [inline client/lib/comment-store/comment-store.js]: :+1: good plan.

File: client/reader/comments/post-comment-content.jsx (hunk 4 above)
Comment (bluefuton) [inline client/reader/comments/post-comment-content.jsx]: :+1:

File: client/lib/comment-store/comment-store.js (hunk 3 above)
Comment (bluefuton) [inline client/lib/comment-store/comment-store.js]: Added in 22a15c4.

File: client/lib/comment-store/test/comment-store-test.js [hunk 5]
```diff
@@ -118,4 +119,55 @@ describe( 'comment-store', function() {
...
+				parent: {
+					ID: 0
+				}
+			}
+		} );
+
+		const comments = CommentStore.getCommentsForPost( siteId, 124 );
+		expect( comments ).to.be.ok;
+		expect( comments[ 0 ][ 0 ].content ).to.eq( 'First!' );
+		expect( comments[ 0 ][ 0 ].state ).to.eq( CommentStates.COMPLETE );
```
Comment (blowery) [inline client/lib/comment-store/test/comment-store-test.js]: hooray!

**Comment by blowery** [comment]
:+1: :shipit:

**Pull request merged**